src/Features/VisualBasic/Portable/CodeRefactorings/NamedArguments/VisualBasicAddNamedArgumentsCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,87 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.CodeRefactorings.NamedArguments
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.NamedArguments
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicAddNamedArgumentsCodeRefactoringProvider)), [Shared]>
+    Friend Class VisualBasicAddNamedArgumentsCodeRefactoringProvider
+        Inherits AbstractAddNamedArgumentsCodeRefactoringProvider(Of ArgumentSyntax)
+
+        Protected Overrides Function GetNamedArguments(arguments As SeparatedSyntaxList(Of ArgumentSyntax), parameters As ImmutableArray(Of IParameterSymbol)) As IEnumerable(Of ArgumentSyntax)
+            Dim result = ArrayBuilder(Of ArgumentSyntax).GetInstance()
+            For index = 0 To arguments.Count - 1
+                Dim argument = arguments(index)
+                Dim parameter = parameters(index)
+                If argument.IsNamed Then
+                    result.Add(argument)
+                ElseIf argument.IsOmitted Then
+                    Continue For
+                ElseIf parameter.IsParams Then
+                    ' Named arguments cannot match a ParamArray parameter
+                    result.Free()
+                    Return Nothing
+                Else
+                    Dim simpleArgument = DirectCast(argument, SimpleArgumentSyntax)
+                    result.Add(simpleArgument.WithNameColonEquals(NameColonEquals(IdentifierName(parameter.Name))).WithTriviaFrom(simpleArgument.Expression))
+                End If
+            Next
+            
+            Return result.ToArrayAndFree()
+        End Function
+
+        Protected Overrides Function IsCandidate(node As SyntaxNode) As Boolean
+            Select Case node.Kind()
+                Case SyntaxKind.InvocationExpression,
```

Comment:
@balajikris It doesn't depends on the selection per @CyrusNajmabadi [feedback](https://github.com/dotnet/roslyn/pull/9934/#discussion_r57479002). `obj` in the example above is not part of the invocation though, it is a conditional access. I think I should cover this for an invocation (VB) or element binding (C#) in `WhenNotNull` so that it will suggest fix for `obj?[arg]` or `obj?(arg)` when caret is placed on `obj`.